src/transformers/pipelines/base.py

```diff
@@ -1038,7 +1038,12 @@ def check_model_type(self, supported_models: Union[List[str], dict]):
                     else:
                         supported_models_names.append(model.__name__)
             supported_models = supported_models_names
-        if self.model.__class__.__name__ not in supported_models:
+        if "Peft" in self.model.__class__.__name__ and hasattr(self.model, "base_model"):
```

Comment:
Hmmm, checking if `Peft` is in the name seems like a flaky way to check this.

What I'd suggest is adding a more general `is_peft_model` utility under `modeling_utils.py` which is tested (and we check with @younesbelkada is correct) which can then be used everywhere. It should also check `is_peft_available`.

Reply:
yes you can do if the model is an isntance of `Peftxxx` class, however we do support already peft model inference: https://huggingface.co/docs/transformers/peft , users that pass a `Peftxxx` class can be considered as bad intent, as they only need to pass a valid path to an adapter model or first load a peft model using `AutoModelForxxx` and pass that to the pipeline. See for example: https://github.com/huggingface/transformers/blob/b45c0f55e06eb4cb1a6cd59dde30d0cef62604d0/tests/peft_integration/test_peft_integration.py#L441